gt_concordance/gt_concordance.py

```diff
@@ -0,0 +1,519 @@
+                continue
+            # sample has format ID22-P2655_176
+            sample_id = row[1].split('-')[-1]
+            # if the same sample occurs twice in the same file, will be overwriten
+            if sample_id not in genotype_data:
+                genotype_data[sample_id] = {}
+            else:
+                click.echo('Sample {} has been already parsed from another (or this) file. Overwriting'.format(sample_id))
+            # rs positions start from 9 element. hopefully the format won't change
+            for rs_id in header[9:]:
+                rs_index = header.index(rs_id)
+                allele1, allele2 = row[rs_index].split()
+                genotype_data[sample_id][rs_id] = [allele1, allele2]
+
+    output_files = []
+    for sample_id in genotype_data:
+        project_id = sample_id.split('_')[0]
+        # create .gt file for each sample
+        output_path = os.path.join(ANALYSIS_PATH, project_id, 'piper_ngi/03_genotype_concordance')
+        if not os.path.exists(output_path):
+            click.echo('Output path does not exist! Path will be created: {}'.format(output_path))
+            os.makedirs(output_path)
+        output_dir = os.path.join(output_path, sample_id)
+        if not os.path.exists(output_dir):
+            os.makedirs(output_dir)
+        filename = os.path.join(output_dir, '{}.maf.gt'.format(sample_id))
+        if os.path.exists(filename):
+            click.echo('File {} already exists. Overwriting'.format(filename))
+        with open(filename, 'w+') as output_file:
+            for rs, alleles in genotype_data[sample_id].items():
+                # checking if snps_data contains such position:
+                if rs not in snps_data:
+                    click.echo('rs_position {} not found in snps_file!!'.format(rs))
+                    click.echo('Skipping')
+                    continue
+                chromosome, position, rs_position, reference, alternative = snps_data.get(rs)
+                output_file.write("{} {} {} {} {} {} {}\n".format(chromosome, position, rs, reference, alternative, alleles[0], alleles[1]))
+            output_files.append(filename)
+    click.echo('{} files have been created in {}/<project>/piper_ngi/03_genotype_concordance'.format(len(output_files), ANALYSIS_PATH))
+
+    # Updating Charon
+    for sample in genotype_data:
+        update_charon(sample=sample, status='AVAILABLE')
+
+    # archiving files
+    archived = []
+    for xl_file in xl_files:
+        try:
+            shutil.move(xl_file, XL_FILES_ARCHIVED)
+        except Exception, e:
+            click.echo('Cannot move file {} to {}'.format(xl_file, XL_FILES_ARCHIVED))
+            click.echo(str(e))
+        else:
+            archived.append(xl_file)
+    click.echo('{} .xlsx files have been archived in {}'.format(len(archived), XL_FILES_ARCHIVED))
+
+
+def parse_maf_snps_file(config):
+    SNPS_FILE = config.get('SNPS_FILE')
+    if SNPS_FILE is None:
```

Comment:
maybe to avoid code repetition you might want to add a loop over a list of variables you want to check if they are present in config or you want them inserted manually